etna/packages/etna-js/spec/helpers.js

```diff
@@ -11,15 +11,60 @@ export const stubUrl = ({
   request,
   status = 200,
   headers = {},
-  host = 'http://www.fake.com'
+  host = 'http://localhost'
 }) => {
-  nock(host)
-    [verb](path, request)
-    .reply(status, response, {
-      'Access-Control-Allow-Origin': '*',
-      'Content-type': 'application/json',
-      ...headers
-    });
+  let nocked;
+
+  console.log('Stubbing for', {request, path, verb, response}, '\nactive')
+
+  return new Promise((resolve, reject) => {
+    const base = nock(host)[verb](path, request);
+    nocked = response instanceof Function
+      ? base.reply(function (uri, body, cb) {
+        response(uri, body, function () {
+          console.log('responding to', { uri, body }, '\nusing custom response cb')
+          nock.removeInterceptor(base);
+          resolve();
+          return cb.apply(this, arguments);
+        })
+      })
+      : base.reply(function () {
+        console.log('responding to', { path, request }, '\nreplying with', { status, response })
+        nock.removeInterceptor(base);
+        resolve();
+        return [status, response, {
+          'Access-Control-Allow-Origin': '*',
+          'Content-type': 'application/json',
+          ...headers
+        }]
+      })
+  });
 };
 
+export async function joinedDeferredPromises(...promiseChains) {
```

Comment:
Joins together chains of test setup promises.  This was useful because I wanted to chain individual request mocks, but to setup those request mocks I needed a promise (for the md5 handling), so I had to hide the inner promises in a function (promises chain by default).  Only really necessary in the deeper tests of uploader.spec.jsx